addons/mail/wizard/mail_compose_message.py

```diff
@@ -518,21 +510,25 @@ def _process_state(self, mail_values_dict):
                 mail_values['failure_type'] = 'mail_bl'
                 # Do not post the mail into the recipient's chatter
                 mail_values['is_notification'] = False
-            # void of falsy values -> error
-            elif not mail_to:
+            # void or falsy values -> error
+            elif not any(recipients['mail_to']):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_missing'
-            elif not mail_to_normalized:
+            elif not any(recipients['mail_to_normalized']):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_invalid'
-            elif optout_emails and mail_to in optout_emails:
+            elif optout_emails and all(
```

Comment:
Same for optout, but less sure about that one.

Reply:
Same a above